src/main/scala/is/hail/rvd/OrderedRVDPartitioner.scala

```diff
@@ -12,24 +12,24 @@ class OrderedRVDPartitioner(
   // rangeBounds: Array[Interval[kType]]
   // rangeBounds is interval containing all keys within a partition
   val rangeBounds: IndexedSeq[Interval],
-  val allowedOverlap: Int
+  allowedOverlap: Int
```

Comment:
Is there a reason why allowedOverlap and partitionKey are two different things?

Reply:
Hopefully my previous response clears this up. As for why I removed the `val`, now that `rangeBounds` can store arbitrary prefixes of the key, no matter what `allowedOverlap`/`partitionKey` is, the `allowedOverlap` parameter expresses purely a condition we want `rangeBounds` to satisfy. (The condition `allowedOverlap` expresses is a bit tricky; I'd be happy to explain in person.) In other words, `allowedOverlap` is used only in an assertion in the constructor, and so is no longer a data-member of the class.